app/black_spots/tasks/get_segments.py

```diff
@@ -0,0 +1,322 @@
+import fiona
+import itertools
+import logging
+import os
+import pytz
+import rtree
+import shutil
+from math import ceil
+import tarfile as t
+import subprocess
+import tempfile
+import multiprocessing
+from shapely.geometry import mapping, shape, LineString, MultiPoint, Point
+from shapely.ops import unary_union
+
+from django.core.files import File
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger()
+
+from black_spots.models import RoadSegmentsShapefile
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+logger = get_task_logger(__name__)
+
+
+INTERSECTION_BUFFER_UNITS = int(os.getenv('INTERSECTION_BUFFER_UNITS', '5'))
+MAX_LINE_UNITS = int(os.getenv('MAX_LINE_UNITS', '200'))
+RECORD_PROJECTION = os.getenv('RECORD_PROJECTION', 'epsg:4326')
+TIMEZONE = pytz.timezone(os.getenv('TIMEZONE', 'Asia/Manila'))
+MATCH_TOLERANCE = int(os.getenv('MATCH_TOLERANCE', '5'))
+RECORD_COL_ID = os.getenv('RECORD_COL_ID', 'record_id')
+RECORD_COL_X = os.getenv('RECORD_COL_X', 'lon')
+RECORD_COL_Y = os.getenv('RECORD_COL_Y', 'lat')
+RECORD_COL_OCCURRED = os.getenv('RECORD_COL_OCCURRED', 'occurred_from')
+RECORD_COL_SEVERE = os.getenv('RECORD_COL_SEVERE', 'Severity')
+RECORD_COL_SEVERE_VALS = os.getenv('RECORD_COL_SEVERE_VALS', 'Fatal,Injury')
+RECORD_COL_PRECIP = os.getenv('RECORD_COL_PRECIP', 'weather')
+RECORD_COL_PRECIP_VALS = os.getenv('RECORD_COL_PRECIP_VALS', 'rain,hail,sleet,snow,thunderstorm,tornado')
+COMBINED_SEGMENTS_SHP_NAME = os.getenv('COMBINED_SEGMENTS_SHP_NAME', 'combined_segments.shp')
+TILE_MAX_UNITS = int(os.getenv('TILE_MAX_UNITS', '3000'))
+
+
+
+
+@shared_task
+def get_segments_shp(path_to_roads_shp):
+    """Save segments to a shapefile and save in model
+
+    Args:
+    :param path_to_roads_shp: (str) path to shapefile
+    """
+    try:
+        logger.info('Reading Roads Shapefile: %s', path_to_roads_shp)
+        roads, road_projection, road_bounds = read_roads(path_to_roads_shp)
+
+        logger.info('Obtaining intersection buffers')
+        int_buffers = get_intersection_buffers(roads, road_bounds, INTERSECTION_BUFFER_UNITS, TILE_MAX_UNITS)
+
+        logger.info('Obtaining intersections')
+        int_multilines, non_int_lines = get_intersection_parts(roads, int_buffers, MAX_LINE_UNITS)
+        combined_segments = int_multilines + non_int_lines
+
+        logger.info('Writing shapefile segments')
+        shp_output_dir = tempfile.mkdtemp()
+        tar_output_dir = tempfile.mkdtemp()
+
+        segments_shp_path = os.path.join(shp_output_dir, COMBINED_SEGMENTS_SHP_NAME)
+        write_segments_shp(segments_shp_path, road_projection, combined_segments)
+        tarball_path = create_tarball(shp_output_dir, tar_output_dir)
```

Comment:
The tarball that's created has a directory structure of: /tmp/tmpXXXXXX/combined_segments.*
I think it'd be good to remove those two levels of temp directories.